### PR TITLE
codegen typescript coin type and protobuf messages

### LINKED ISSUE #1019: Publish wallet core types package (typescript)
It will contain `CoinType` and protobuf models

Host it here: https://github.com/trustwallet/wallet-core/packages

### PR DESCRIPTION
**Pull request by hewigovens**
Fixes #1019 

Changes:

- Codegen `CoinType` from coins.json (tweak code form old web-core), codegen proto messages (currently only Ethereum.proto)
- Add typescript CI and tests
- [x] Auto publish

### CODE REVIEW

File: coins.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
         "symbol": "NAS",
         "decimals": 18,
         "blockchain": "Nebulas",
+        "slip44": 2718,
```
Comment (catenocrypt) [inline coins.json:7]: Some pondering:
- Duplication.  It is good that this is a separate field now, it is easier to handle.  But duplication (present in slip44 and in derivation path) can be bad, can diverge, which can cause trouble.  OTOH, it allows purposefully different values (I have in mind subchains).
- Naming.  I know this is slip44 ID, but in most cases we just refer to it as coin type / coin ID.  I would call it `numID`, and in some special cases even deviate from slip44 (e.g. subchains).

Comment (hewigovens) [reply coins.json:7]: We have a string type id tho, I can rename slip44 to coinId, slip44 is not good variable

File: .github/workflows/ts-ci.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        node-version: ${{ matrix.node-version }}
+    - run: yarn install
+      working-directory: typescript
+    - run: yarn build
+      working-directory: typescript
+    - run: yarn test
+      working-directory: typescript
+    - name: Publish
+      run: |
+        echo "//npm.pkg.github.com/:_authToken=${TOKEN}" >> .npmrc
```
Comment (vikmeup) [inline .github/workflows/ts-ci.yml:29]: Is there better ways to set env var for npm?

File: coins.json (hunk 1 above)
Comment (vikmeup) [reply coins.json:7]: Can we delete this value and read it from derivation path? It's not worth duplicating

Comment (vikmeup) [reply coins.json:7]: This might be confusing and will lead to bugs

Comment (hewigovens) [reply coins.json:7]: ok, let me revert it, maybe it's time to change to string based id? in some cases, slip44 id is not equal to derivation path

File: .github/workflows/ts-ci.yml (hunk 2 above)
Comment (hewigovens) [reply .github/workflows/ts-ci.yml:29]: another way is npm login, seems not working in CI (interactive)

File: coins.json (hunk 1 above)
Comment (vikmeup) [reply coins.json:7]: In which coins slip44 not equal to deviation path? 

We can do slip:44

And then derivation path: `m/{{slip}}/0/`

As an option

**Pull request merged**